### PR TITLE
feat: add shape

### PR DESCRIPTION
**Pull request by r0ohafza**
This PR adds support for `Shape` network.

### CODE REVIEW

File: packages/splits-kit/storybook/constants/chains.ts [hunk 1]
```diff
@@ -141,4 +142,12 @@ export const STORYBOOK_CHAIN_INFO: ChainInfo = {
       `https://base-mainnet.g.alchemy.com/v2/${process.env.STORYBOOK_ALCHEMY_API_KEY}`,
     ],
   },
+  [shape.id]: {
+    ...CHAIN_INFO[shape.id],
+    viemChain: shape,
+    rpcUrls: [
+      // TODO: rpc url
```
Comment (mihoward21) [inline packages/splits-kit/storybook/constants/chains.ts:149]: lol, why do we have this on every single one...

File: packages/splits-sdk-react/src/hooks/data.ts [hunk 2]
```diff
@@ -160,8 +160,9 @@ export const useSplitMetadataViaProvider = (
             currentBlockNumber > splitV1StartBlock &&
             currentBlockNumber > splitV2StartBlock
           ) {
-            const startBlock =
+            const rangeStart =
               currentBlockNumber - blockRange * BigInt(LOGS_SEARCH_BATCH_SIZE)
+            const startBlock = rangeStart >= BigInt(0) ? rangeStart : BigInt(0)
```
Comment (mihoward21) [inline packages/splits-sdk-react/src/hooks/data.ts:165]: this was the issue you were running into, yea?

File: packages/splits-sdk/src/constants/index.ts [hunk 3]
```diff
@@ -173,6 +175,7 @@ export const ALL_CHAIN_IDS = [
   ...ZORA_CHAIN_IDS,
   ...BASE_CHAIN_IDS,
   ...BLAST_CHAIN_IDS,
+  ...SHAPE_CHAIN_IDS,
 ]
 
 export const SPLITS_SUPPORTED_CHAIN_IDS = [3, 4, 42, ...ALL_CHAIN_IDS]
```
Comment (mihoward21) [inline packages/splits-sdk/src/constants/index.ts:181]: is v1 on shape? or just v2? i think these changes will add it to this v1 supported list, right?

**Review by mihoward21 — COMMENTED** [review]
this all looks good. just double checking if shape is on v1

File: packages/splits-sdk/src/constants/index.ts (hunk 3 above)
Comment (r0ohafza) [reply packages/splits-sdk/src/constants/index.ts:181]: oh I see, v1 is not on shape, reverting this change

File: packages/splits-kit/storybook/constants/chains.ts (hunk 1 above)
Comment (r0ohafza) [reply packages/splits-kit/storybook/constants/chains.ts:149]: lol

File: packages/splits-sdk-react/src/hooks/data.ts (hunk 2 above)
Comment (r0ohafza) [reply packages/splits-sdk-react/src/hooks/data.ts:165]: yes

File: packages/splits-sdk/src/constants/index.ts (hunk 3 above)
Comment (mihoward21) [reply packages/splits-sdk/src/constants/index.ts]: does just removing it from there work though? unfortunately i think that might block it from being used in the sdk? i see we pass ALL_CHAIN_IDS into the `supportedChainIds` field in a couple places.

Might need to just hardcode this `SPLITS_SUPPORTED_CHAIN_IDS` list (given it won't change anymore). And then all chain ids does include the new ones?

**Review by mihoward21 — APPROVED** [review]
cool, thank you getting all this in

**Pull request merged**